### PR TITLE
Add action for generating onesignal apps

### PR DESCRIPTION
**Pull request by timothybarraclough**

### CODE REVIEW

File: lib/fastlane/actions/onesignal.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          raise "Unexpected #{response.code} with response: #{response.body}".red
+        end
+      end
+
+      def self.description
+        "Create a new OneSignal application"
+      end
+
+      def self.details
+        # Optional:
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please remove this comment

File: lib/fastlane/actions/onesignal.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        "You can use this action to automatically create a OneSignal application. You can also upload a .p12 with password, a GCM key, or both"
+      end
+
+      def self.available_options
+        # Define all options your action supports.
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please remove this comment

File: lib/fastlane/actions/onesignal.rb [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          FastlaneCore::ConfigItem.new(key: :apns_env,
+                                       env_name: "APNS_ENV",
+                                       description: "APNS environment",
+                                       optional: true,
+                                       default_value: 'production')
+        ]
+      end
+
+      def self.output
+        # Define the shared values you are going to provide
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please remove this comment

File: lib/fastlane/actions/onesignal.rb [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        [
+          ['ONE_SIGNAL_APP_ID', 'The OneSignal app ID of the newly created app'],
+          ['ONE_SIGNAL_APP_AUTH_KEY', 'The auth token for the newly created OneSignal app']
+        ]
+      end
+
+      def self.return_value
+        # If you method provides a return value, you can describe here what it does
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please remove this comment

File: lib/fastlane/actions/onesignal.rb [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please remove this comment

File: lib/fastlane/actions/onesignal.rb [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        ["timothybarraclough / smartshowltd"]
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: Please change to 
```ruby
["timothybarraclough", "smartshowltd"]
```

File: docs/Actions.md [hunk 7]
```diff
@@ -1701,6 +1701,22 @@ podio_item(
 ```
 To see all environment values, please run ```fastlane action podio_item```.
 
+### [OneSignal](http://onesignal.com)
```
Comment (KrauseFx) [inline docs/Actions.md]: Should be https

**Comment by KrauseFx** [comment]
Hey @timothybarraclough, thanks for this pull request, this looks like a great addition to fastlane. I added some comments before it's ready to be merged :+1:

**Comment by timothybarraclough** [comment]
Hey @KrauseFx, thanks for that. I've removed the extraneous comments and updated the docs accordingly.

File: lib/fastlane/actions/onesignal.rb [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+          ['ONE_SIGNAL_APP_ID', 'The OneSignal app ID of the newly created app'],
+          ['ONE_SIGNAL_APP_AUTH_KEY', 'The auth token for the newly created OneSignal app']
+        ]
+      end
+
+      def self.return_value
+      end
+
+      def self.authors
+        ["timothybarraclough, smartshowltd"]
```
Comment (KrauseFx) [inline lib/fastlane/actions/onesignal.rb]: It should be 
```ruby
["timothybarraclough", "smartshowltd"]
```

**Comment by KrauseFx** [comment]
One last thing, thanks for updating the PR :+1:

**Comment by timothybarraclough** [comment]
:+1:

**Comment by KrauseFx** [comment]
Please squash the commits to one (https://github.com/fastlane/countdown#submit-a-pull-request) and then we can merge :+1: Thanks again for all your contribution

**Comment by timothybarraclough** [comment]
All Done! Thanks.

**Comment by KrauseFx** [comment]
Perfect, thank you @timothybarraclough :+1:

**Pull request merged**